### PR TITLE
Refactor to depend on new sbi 0.20.0

### PR DESCRIPTION
**Pull request by janfb**
refactor `sbi` run scripts to match the new API of `sbi` version `>=0.20.0`. 

- [x] depend on newest `sbi` version 0.20.0 to support passing `TransformedDistributions` as prior
- [x] run all tests.

### CODE REVIEW

**Comment by janfb** [comment]
ready for review @jan-matthis

**Review by jan-matthis — COMMENTED** [review]
Looks great, thank you!

A few small comments below

File: sbibm/algorithms/sbi/snle.py [hunk 1]
```diff
@@ -106,8 +109,8 @@ def run(
 
     posteriors = []
     proposal = prior
-    mcmc_parameters["warmup_steps"] = 25
-    mcmc_parameters["enable_transform"] = False  # NOTE: Disable `sbi` auto-transforms, since `sbibm` does its own
+    # NOTE: why was this hard coded here?
+    # mcmc_parameters["warmup_steps"] = 25
```
[the remark this answers is not in the document]
Comment (janfb) [reply sbibm/algorithms/sbi/snle.py:113]: it's already contained in the `mcmc_parameters` init

**Comment by janfb** [comment]
Thanks for the feedback! 

I remove a couple of test files and refactored them into one single `sbi` test file: `test_sbi_run_methods.py`

That tests now also inclused `sl`.

**Comment by janfb** [comment]
I also changed the requirement to `sbi>=0.20.0`, as we will likely release a new version that incluces BNRE and SNRE_C soon. 

Or do you prefer to fix the setup to one specific `sbi` version?

File: tests/algorithms/sbi/test_snre_posterior.py [hunk 2]
```diff
@@ -1,47 +0,0 @@
-import pytest
```
[the remark this answers is not in the document]
Comment (janfb) [reply tests/algorithms/sbi/test_snre_posterior.py:1]: moved into `pytest.parametrize`

**Comment by jan-matthis** [comment]
Thanks!

**Pull request merged**